### PR TITLE
Update allocation_number.py

### PR DESCRIPTION
**Pull request by wind-Lv**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by mrmaxguns** [comment]
Ah... It looks like black has exited with 1. Make sure to run black before a pull request so that it can properly format files.

**Comment by wind-Lv** [comment]
@mrmaxguns Please review again.

**Comment by mrmaxguns** [comment]
Looks good to me. I believe @cclauss can merge this pull request.

File: maths/allocation_number.py [hunk 1]
```diff
@@ -35,7 +34,7 @@ def allocation_num(number_of_bytes: int, partitions: int) -> list[str]:
     bytes_per_partition = number_of_bytes // partitions
     allocation_list = []
     for i in range(partitions):
-        start_bytes = i * bytes_per_partition + 1
+        start_bytes = 0 if i * bytes_per_partition == 0 else i * bytes_per_partition + 1
```
Comment (cclauss) [inline maths/allocation_number.py:37]: This makes the first block larger than all other blocks and allocates one block too many.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: maths/allocation_number.py [hunk 2]
```diff
@@ -15,9 +15,9 @@ def allocation_num(number_of_bytes: int, partitions: int) -> list[str]:
     :param partitions: the number of partition need to be allocated.
     :return: list of bytes to be assigned to each worker thread
     >>> allocation_num(16647, 4)
-    ['0-4161', '4162-8322', '8323-12483', '12484-16647']
+    ['0-4161', '4162-8323', '8324-12485', '12486-16647']
     >>> allocation_num(50000, 5)
-    ['0-10000', '10001-20000', '20001-30000', '30001-40000', '40001-50000']
+    ['0-9999', '10000-19999', '20000-29999', '30000-39999', '40000-50000']
```
Comment (cclauss) [inline maths/allocation_number.py:20]: 49999 or the function is still allocating one byte too many.

Comment (wind-Lv) [reply maths/allocation_number.py:20]: Some number can't be divisible.This is normal.

**Comment by cclauss** [comment]
I do not see the benefit of these changes.

**Comment by wind-Lv** [comment]
In requests, the Range must be from 0.

**Comment by Erdal-web** [comment]
I need a python app for apple Ipad 
But free application 
Who advice app to me 

iPhone’umdan gönderildi

**Pull request closed without merging**